front/production_checks/checks/managed_data_sources_gc.ts

```diff
@@ -0,0 +1,186 @@
+    if (coreDs.length === 0) {
+      await reportFailure(
+        { frontDataSourceId: ds.id },
+        `No core data source found for front data source`
+      );
+      continue;
+    }
+    const coreDocumentsData = await core_sequelize.query(
+      `SELECT id, document_id, parents FROM data_sources_documents WHERE "data_source" = :coreDsId AND status = 'latest'`,
+      {
+        replacements: {
+          coreDsId: coreDs[0].id,
+        },
+        type: QueryTypes.SELECT,
+      }
+    );
+
+    const coreDocuments = coreDocumentsData as CoreDSDocument[];
+    const checkFiles =
+      CHECK_FILES_BY_TYPE[ds.connectorProvider as ConnectorProvider];
+    if (checkFiles) {
+      const result = await checkFiles(
+        ds.connectorId,
+        coreDocuments,
+        connectorsSequelize
+      );
+      if (result.isErr()) {
+        await reportFailure(
+          {
+            frontDataSourceId: ds.id,
+            coreDataSourceId: coreDs[0].id,
+            conncetorId: ds.connectorId,
+            connectorProvider: ds.connectorProvider,
+            errorPayload: result.error,
+          },
+          `Files from managed Data Sources not garbage collected.`
+        );
+      } else {
+        await reportSuccess({ frontDataSourceId: ds.id });
+      }
+    }
+  }
+};
+
+export const CHECK_FILES_BY_TYPE: Record<
+  ConnectorProvider,
+  DocumentsChecker | undefined
+> = {
+  slack: async (
+    connectorId: ModelId,
+    documents: CoreDSDocument[],
+    connectorsSequelize: Sequelize
+  ): Promise<Result<void, unknown>> => {
+    const missingChannels: string[] = [];
+    const coreChannel = new Set(documents.map((d) => d.parents[0]));
+    for (const channelId of coreChannel) {
+      const selectedChannel: { id: number; permission: string }[] =
+        await connectorsSequelize.query(
+          'select id, permission from slack_channels WHERE "slackChannelId" = :channelId and "connectorId" = :connectorId',
+          {
+            replacements: {
+              channelId,
+              connectorId,
+            },
+            type: QueryTypes.SELECT,
+          }
+        );
+      if (selectedChannel.length === 0) {
+        missingChannels.push(channelId);
+      }
+    }
+    if (missingChannels.length) {
+      return new Err({ missingChannels });
+    } else {
+      return new Ok(void 0);
+    }
+  },
+  notion: undefined,
+  github: undefined,
+  google_drive: async (
+    connectorId: ModelId,
+    documents: CoreDSDocument[],
+    connectorsSequelize: Sequelize
+  ): Promise<Result<void, unknown>> => {
+    const results: CoreDSDocument[] = [];
+    do {
+      const slice = documents.splice(0, 100);
+      const dustFileIds = slice.map((d) => d.document_id);
+
+      const documentsData = await connectorsSequelize.query(
+        'SELECT id as "googleDriveFileId", "dustFileId" FROM google_drive_files WHERE "dustFileId" IN (:documentIds) and "connectorId" = :connectorId',
```

Comment:
I think we should stick with the flow that was used in the migration

Get the files create a hash, get all core files (we can limit to document_id to make it more lightweight), check them all